### PR TITLE
refactor: Remove EDV from BDD tests

### PR DESCRIPTION
**Pull request by ypukhta**
Remove EDV support from KMS BDD tests

Signed-off-by: Yevgen Pukhta <eugene.pukhta@gmail.com>

### CODE REVIEW

**Review by rolsonquadras — APPROVED** [review]

File: test/bdd/pkg/kms/auth_edv_steps.go [hunk 1]
```diff
@@ -7,40 +7,19 @@ SPDX-License-Identifier: Apache-2.0
...
-	"github.com/trustbloc/kms/test/bdd/pkg/internal/cryptoutil"
-)
-
-const (
-	edvBasePath    = "/encrypted-data-vaults"
-	secretEndpoint = "/secret"
 )
 
-func (s *Steps) storeSecretInHubAuth(userName string) error {
+func (s *Steps) loginIntoAuthServer(userName string) error {
```
Comment (DRK3) [inline test/bdd/pkg/kms/auth_edv_steps.go:17]: Typo here - I think you mean `logInToAuthServer`

File: test/bdd/pkg/kms/kms_steps.go [hunk 2]
```diff
@@ -140,27 +133,12 @@ func (s *Steps) createKeystoreAndKey(user, keyType string) error {
...
-	edvCapability, err := s.createChainCapability(u)
-	if err != nil {
-		return err
-	}
-
-	capabilityBytes, err := json.Marshal(edvCapability)
-	if err != nil {
-		return err
-	}
+	//if err := s.createDID(u); err != nil {
```
Comment (DRK3) [inline test/bdd/pkg/kms/kms_steps.go:136]: Did you meant to leave this here?

File: test/bdd/pkg/kms/zcap.go [hunk 3]
```diff
@@ -39,133 +27,32 @@ type signer interface {
 	Alg() string
 }
 
-type authzKMSSigner struct {
+type zCapAuthUserSigner struct {
```
Comment (DRK3) [inline test/bdd/pkg/kms/zcap.go:30]: Should be `zcapAuthUserSigner` not `zCapAuthUserSigner`

File: test/bdd/pkg/kms/zcap.go [hunk 4]
```diff
@@ -39,133 +27,32 @@ type signer interface {
...
 	s         *Steps
 	authzUser *user
 }
 
-func newAuthzKMSSigner(s *Steps, authzUser *user) *authzKMSSigner {
-	return &authzKMSSigner{s: s, authzUser: authzUser}
+func newZCapAuthUserSigner(s *Steps, authzUser *user) *zCapAuthUserSigner {
```
Comment (DRK3) [inline test/bdd/pkg/kms/zcap.go:35]: Should be `newZCAPAuthUserSigner` not `newZCapAuthUserSigner`

File: test/bdd/pkg/kms/zcap.go (hunk 3 above)
Comment (DRK3) [reply test/bdd/pkg/kms/zcap.go:30]: (`ZCAP` is an acronym, so should be in all caps, except when at the beginning of an unexported name, in which case it's all lowercase)

File: test/bdd/pkg/internal/httputil/httputil.go [hunk 5]
```diff
@@ -161,13 +161,6 @@ func WithBody(val []byte) Opt {
 	}
 }
 
-// WithBearerToken specifies an authorization bearer token.
-func WithBearerToken(token string) Opt {
-	return func(o *options) {
-		o.bearerToken = token
-	}
-}
-
```
Comment (aholovko) [inline test/bdd/pkg/internal/httputil/httputil.go:170]: if this is not needed, remove plz `bearerToken` from `options` struct and behavior related to setting `Authorization` header in `DoRequest` method

File: test/bdd/pkg/internal/httputil/httputil.go [hunk 6]
```diff
@@ -181,10 +174,3 @@ func WithSigner(signer requestSigner) Opt {
 		o.signer = signer
 	}
 }
-
-// WithParsedResponse specifies type to unmarshal response body.
-func WithParsedResponse(r interface{}) Opt {
```
Comment (aholovko) [inline test/bdd/pkg/internal/httputil/httputil.go:186]: consider cleaning up `parsedResponse`-related behavior as well (the same considerations as for `bearerToken` case above)

**Review by aholovko — APPROVED** [review]

File: test/bdd/pkg/kms/auth_edv_steps.go (hunk 1 above)
Comment (ypukhta) [reply test/bdd/pkg/kms/auth_edv_steps.go]: fixed

File: test/bdd/pkg/kms/kms_steps.go (hunk 2 above)
Comment (ypukhta) [reply test/bdd/pkg/kms/kms_steps.go]: removed

File: test/bdd/pkg/kms/zcap.go (hunk 3 above)
Comment (ypukhta) [reply test/bdd/pkg/kms/zcap.go]: fixed

File: test/bdd/pkg/kms/zcap.go (hunk 4 above)
Comment (ypukhta) [reply test/bdd/pkg/kms/zcap.go]: fixed

File: test/bdd/pkg/internal/httputil/httputil.go (hunk 5 above)
Comment (ypukhta) [reply test/bdd/pkg/internal/httputil/httputil.go:170]: removed

File: test/bdd/pkg/internal/httputil/httputil.go (hunk 6 above)
Comment (ypukhta) [reply test/bdd/pkg/internal/httputil/httputil.go:186]: removed

**Pull request merged**